Software/Development/SemesterProject/App/errorManager.c

```diff
@@ -49,19 +49,38 @@ static errorStatus systemErrorStatus = NO_ERROR;
 
 /*
  * This function updates the the error status of the different measurements,
- * and determines if safety reactions should be performed
+ * and determines if safety reactions should be performed.
+ *
+ * The error monitoring is divided into high and low priority signals, to minimize
+ * the execution time.
  */
-void monitorErrorSources(void)
+void monitorHighPriorityErrorSources(void)
 {
-    errorStatusList.currentErrorStatus = areCurrentMeasurementsWithinThresholds();
-    errorStatusList.batteryErrorStatus = areBatteryMeasurementsWithinThresholds();
-    errorStatusList.adcErrorStatus = areAdcMeasurementsWithinThresholds();
+    Uint16 errorStatus = getHighPriorityErrorStatus();
+
+    errorStatusList.currentErrorStatus = areCurrentMeasurementsWithinThresholds(errorStatus);
 
     if (errorStatusList.currentErrorStatus == ERROR_HAS_HAPPENED)
     {
         performSafetyReactions();
         updateCurrentErrorStatus();
     }
+}
+
+void performHighPriorityErrorMonitoring(void)
+{

```

Comment:
Why no need to wait for filter to settle?

Reply:
If we dont wait for the filters to settle in the during startup, we will generate some wrong errors.